cmd/agent/main.go

```diff
@@ -42,33 +43,72 @@ const (
 )
 
 func main() {
-	if _, err := reconcileKubelet(context.Background()); err != nil {
-		logrus.Warnf("failed to reconcile kubelet, error: %v", err)
-	}
+	var err error
 
-	logrus.SetOutput(colorable.NewColorableStdout())
-	logserver.StartServerWithDefaults()
-	if os.Getenv("CATTLE_DEBUG") == "true" || os.Getenv("RANCHER_DEBUG") == "true" {
-		logrus.SetLevel(logrus.DebugLevel)
-	}
+	if len(os.Args) > 1 {
+		err = runArgs()
+	} else {
+		if _, err := reconcileKubelet(context.Background()); err != nil {
+			logrus.Warnf("failed to reconcile kubelet, error: %v", err)
+		}
 
-	initFeatures()
+		logrus.SetOutput(colorable.NewColorableStdout())
+		logserver.StartServerWithDefaults()
+		if os.Getenv("CATTLE_DEBUG") == "true" || os.Getenv("RANCHER_DEBUG") == "true" {
+			logrus.SetLevel(logrus.DebugLevel)
+		}
 
-	var err error
+		initFeatures()
 
-	if os.Getenv("CLUSTER_CLEANUP") == "true" {
-		err = clean.Cluster()
-	} else if os.Getenv("BINDING_CLEANUP") == "true" {
-		err = clean.Bindings(nil)
-	} else {
-		err = run()
+		if os.Getenv("CLUSTER_CLEANUP") == "true" {
+			err = clean.Cluster()
+		} else if os.Getenv("BINDING_CLEANUP") == "true" {
+			err = clean.Bindings(nil)
+		} else {
+			err = run()
+		}
 	}
 
 	if err != nil {
 		logrus.Fatal(err)
 	}
 }
 
+func runArgs() error {
+	switch os.Args[1] {
+	case "clean":
+		return runClean()
+	default:
+		return run()
+	}
+}
+
+func runClean() error {
+	if len(os.Args) > 2 {
```

Comment:
nit: Is it valid to pass in more than 2 `os.Args`? If not, we should return an error.

Reply:
👍